### PR TITLE
NO-JIRA: hypershift: implements missing client calls needed for e2e tests 

### PR DESCRIPTION
**Pull request by Tal-or**
This PR has been split  from #1084 and is a prerequisite for it.
It implements missing API calls that are needed during the e2e tests adjustments

Co-authored-by: Ronny Baturov <rbaturov@redhat.com>
Signed-off-by: Talor Itzhak <titzhak@redhat.com>

### CODE REVIEW

**Comment by Tal-or** [comment]
/hold 
Depend on https://github.com/openshift/cluster-node-tuning-operator/pull/1100

**Review by ffromani — COMMENTED** [review]
/approve

**Comment by Tal-or** [comment]
/hold cancel
https://github.com/openshift/cluster-node-tuning-operator/pull/1100 got merged

File: test/e2e/performanceprofile/functests/utils/client/dataplane.go [hunk 1]
```diff
@@ -14,55 +14,55 @@ type dataPlaneImpl struct {
 }
 
 func (dpi *dataPlaneImpl) Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
-	if hypershift.IsEncapsulatedInConfigMap(obj) {
+	if hypershift.GetObjectConfigMapDataKey(obj) != "" {
```
Comment (ffromani) [inline test/e2e/performanceprofile/functests/utils/client/dataplane.go:17]: keep the intent for better readability, reimplement the function:
```
func IsEncapsulatedInConfigMap(obj runtime.Object) bool {
    return GetObjectConfigMapDataKey(obj) != ""
}
```

Comment (ffromani) [reply test/e2e/performanceprofile/functests/utils/client/dataplane.go:17]: or change again `GetObjectConfigMapDataKey(obj)` so it returns the key and a bool (which feels redundant) and use it like
```
if _, ok := hypershift.GetObjectConfigMapDataKey(obj); !ok {
```
this is again to convey the intent more explicitely

**Review by ffromani — COMMENTED** [review]
/approve

PTAL to the inline comments

File: test/e2e/performanceprofile/functests/utils/client/dataplane.go (hunk 1 above)
Comment (Tal-or) [reply test/e2e/performanceprofile/functests/utils/client/dataplane.go:17]: I liked the first idea, it's great, thanks

**Pull request merged**